### PR TITLE
fix create pull request from base repo to forked by api

### LINKED ISSUE #20475: Pull request creation to forked repository via web api fails
### Description

Gitea prevents pull request creation with the web api from the original repository to the fork with the error message: `<Response [404]> b'{"errors":null,"message":"GetForkedRepo","url":"https://try.gitea.io/api/swagger"}\n'`.

I managed to reproduce the issue on https://try.gitea.io/Lixfel/FreeSql .

Reproduction code:
```
token = "?"
import requests
response = requests.post(
        f'https://try.gitea.io/api/v1/repos/Lixfel/FreeSql/pulls?token={token}',
        json={
            "base": "master",
            "head": "huihuiyule:master",
            "title": "Update Upstream"
        })
print(f"{response} {response.content}")
```

Bug origin assumption:
https://github.com/go-gitea/gitea/blob/main/routers/api/v1/repo/pull.go#L967 seems to only allow PRs from the fork to the original repository, a method call `GetForkingRepo` is missing.

### Gitea Version

1.16.9

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

https://gist.github.com/lixfel/e0c36e3b85a1b8ffbf994baf275e49bd

### Screenshots

_No response_

### Git Version

1:2.30.2-1 (apt version)

### Operating System

Debian

### How are you running Gitea?

Use of Gitea Downloads site; execution with systemd

### Database

MySQL

### PR DESCRIPTION
**Pull request by a1012112796**
fix #20475

### CODE REVIEW

**Comment by noerw** [comment]
Don't we have this logic already? Seems odd to repeat this for the API

**Comment by a1012112796** [comment]
> Don't we have this logic already? Seems odd to repeat this for the API

yes, which is used to handle reqeust from web ui, but it's not a easy job to reuse it ...

https://github.com/go-gitea/gitea/blob/4fc53a3f30f6c71325b01654019c28c334271109/routers/web/repo/compare.go#L340-L353

File: routers/api/v1/repo/pull.go [hunk 1]
```diff
@@ -966,8 +966,33 @@ func parseCompareInfo(ctx *context.APIContext, form api.CreatePullRequestOption)
...
+	var rootRepo *repo_model.Repository
+	if baseRepo.IsFork && !isSameRepo {
+		err = baseRepo.GetBaseRepo()
+		if err != nil {
+			if !repo_model.IsErrRepoNotExist(err) {
+				ctx.ServerError("Unable to find root repo", err)
+				return nil, nil, nil, nil, "", ""
+			}
+		} else {
+			rootRepo = baseRepo.BaseRepo
```
Comment (lunny) [inline routers/api/v1/repo/pull.go:988]: Should we still find `BaseRepo`'s base repository?

Comment (a1012112796) [reply routers/api/v1/repo/pull.go:988]: Hmm, that's so old that I has forgoten why ...,  maybe it's just be copyed from here ...
https://github.com/go-gitea/gitea/blob/e81ccc406bf723a5a58d685e7782f281736affd4/routers/web/repo/compare.go#L340-L351
If you think it's wrong, this pr can be closed, and looks need some refactor ...

**Comment by denyskon** [comment]
What is the state of this PR? Is the issue still present, and if so, is this a valid solution?

**Pull request closed without merging**